Actions/PullPowerPlatformChanges/action.yaml

```diff
@@ -0,0 +1,150 @@
+      shell: ${{ inputs.shell }}
+      env:
+        _actionsRepo: ${{ inputs.actionsRepo }}
+      run: |
+        # Using hardcoded reference to microsoft/AL-Go-Actions inside composote actions doesn't work with preview nor direct development branches
+        # For this reason, we get the actionsRepo from the calling workflow (which might be freddydk/AL-Go@PPP or microsoft/AL-Go@sha or microsoft/AL-Go-Actions@v4.1 )
+        # Download actions to ./_AL-Go/Actions
+        $actionsRef = $env:_actionsRepo.Split('@')[1]
+        $actionsRepo = $env:_actionsRepo.Split('@')[0]
+        $actionsPath = './_AL-Go/Actions'
+        if ($actionsRepo -like '*/Actions') {
+          $actionsRepo = $actionsRepo.SubString(0,$actionsRepo.Length-8)
+          $actionsPath = './_AL-Go'
+        }
+        Add-Content -encoding utf8 -path $env:GITHUB_ENV -value "actionsRepo=$actionsRepo"
+        Add-Content -encoding utf8 -path $env:GITHUB_ENV -value "actionsRef=$actionsRef"
+        Add-Content -encoding utf8 -path $env:GITHUB_ENV -value "actionsPath=$actionsPath"
+
+    - name: Check out AL-Go Actions
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ env.actionsRepo }}
+        ref: ${{ env.actionsRef }}
+        path: ${{ env.actionsPath }}
+
+    - name: Parse DeployToSettings and AuthContext
+      id: ReadPowerPlatformSettings
+      uses: ./_AL-Go/Actions/ReadPowerPlatformSettings
+      with:
+        shell: ${{ inputs.shell }}
+        deploymentEnvironmentsJson: ${{ inputs.deploymentEnvironmentsJson }} 
+        environmentName: ${{ inputs.environmentName }}
+
+    - name: Who am I (username)
+      if: steps.ReadPowerPlatformSettings.outputs.ppUserName != ''
+      uses: microsoft/powerplatform-actions/who-am-i@v1
+      with:
+        environment-url: ${{ steps.ReadPowerPlatformSettings.outputs.ppEnvironmentUrl}}
+        user-name: ${{ steps.ReadPowerPlatformSettings.outputs.ppUserName }}
+        password-secret: ${{ steps.ReadPowerPlatformSettings.outputs.ppPassword }}
+
+    - name: Who am I (application ID)
+      if: steps.ReadPowerPlatformSettings.outputs.ppApplicationId != ''
+      uses: microsoft/powerplatform-actions/who-am-i@v1
+      with:
+        environment-url: ${{ steps.ReadPowerPlatformSettings.outputs.ppEnvironmentUrl }}
+        tenant-id: ${{ steps.ReadPowerPlatformSettings.outputs.ppTenantId }}
+        app-id: ${{ steps.ReadPowerPlatformSettings.outputs.ppApplicationId }}
+        client-secret: ${{ steps.ReadPowerPlatformSettings.outputs.ppClientSecret }}
+
+    - name: Set up new branch for changes
+      shell: ${{ inputs.shell }}
+      env:
+        _actor: ${{ inputs.actor }}
+        _token: ${{ inputs.token }}
+        _updateBranch: ${{ inputs.updateBranch }}
+        _directCommit: ${{ inputs.directCommit }}
+      run: ${{ github.action_path }}\GitCloneReponsitory.ps1 -actor $ENV:_actor -token $ENV:_token -updateBranch $ENV:_updateBranch -directCommit ($ENV:_directCommit -eq 'true')
```

Comment:
When running script in actions, we wrap the script call like:

```
        $errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
        try {
          ${{ github.action_path }}/some-script.ps1
        }
        catch {
          Write-Host "::ERROR::Unexpected error when running action. Error Message: $($_.Exception.Message.Replace("`r",'').Replace("`n",' ')), StackTrace: $($_.ScriptStackTrace.Replace("`r",'').Replace("`n",' <- '))";
          exit 1
        }
```

There was some reasons for that and the one on top of my mind is that some failures do not stop the action/workflow. @freddydk you probably remember more.

Reply:
correct